packages/rich-text/src/component/index.native.js

```diff
@@ -1248,13 +1260,15 @@ export default compose( [
 
 		const settings = getSettings();
 		const baseGlobalStyles = settings?.__experimentalGlobalStylesBaseStyles;
-		const experimentalFeatures =
-			settings?.__experimentalFeatures?.color?.palette;
-		const colorPalette =
-			experimentalFeatures?.user ??
-			experimentalFeatures?.theme ??
-			experimentalFeatures?.default ??
-			settings?.colors;
+		const colorsPalettes = settings?.__experimentalFeatures?.color?.palette;
+		const allColorsPalette = [
+			...( colorsPalettes?.theme ? colorsPalettes.theme : [] ),
+			...( colorsPalettes?.custom ? colorsPalettes.custom : [] ),
+			...( colorsPalettes?.default ? colorsPalettes.default : [] ),
```

Comment:
Nitpick: I think we could simplify these lines.
```suggestion
			...( colorsPalettes?.theme || [] ),
			...( colorsPalettes?.custom || [] ),
			...( colorsPalettes?.default || [] ),
```

Reply:
Thanks for the suggestion, I applied those changes 👍